From 8e3083375f9e39112e23f32ac0ac5afd98bb2d64 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:25 -0400
Subject: [fs] btrfs: don't check nodes for extent items

Message-id: <1407179109-13714-379-git-send-email-sandeen@redhat.com>
Patchwork-id: 87317
O-Subject: [RHEL7.1 PATCH 378/422] Btrfs: don't check nodes for extent items
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Josef Bacik <jbacik@fb.com>

commit 8a56457f5f8fa7c2698ffae8545214c5b96a2cb5
Author: Josef Bacik <jbacik@fb.com>
Date:   Thu Jun 5 16:08:45 2014 -0400

    Btrfs: don't check nodes for extent items

    The backref code was looking at nodes as well as leaves when we tried to
    populate extent item entries.  This is not good, and although we go away with it
    for the most part because we'd skip where disk_bytenr != random_memory,
    sometimes random_memory would match and suddenly boom.  This fixes that problem.
    Thanks,

    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/backref.c b/fs/btrfs/backref.c
index f09aa18..84d0912 100644
--- a/fs/btrfs/backref.c
+++ b/fs/btrfs/backref.c
@@ -988,11 +988,12 @@ again:
     goto out;
   }
   if (ref->count && ref->parent) {
-   if (extent_item_pos && !ref->inode_list) {
+   if (extent_item_pos && !ref->inode_list &&
+       ref->level == 0) {
     u32 bsz;
     struct extent_buffer *eb;
     bsz = btrfs_level_size(fs_info->extent_root,
-       info_level);
+       ref->level);
     eb = read_tree_block(fs_info->extent_root,
           ref->parent, bsz, 0);
     if (!eb || !extent_buffer_uptodate(eb)) {
-- 
1.7.1